[sarahteonin] iP

sarahteonin opened the pull request:
### Bob is your friendly companion to help you keep track of your tasks!
> “Your tasks are just steps to your goals. Stay organized, and you’ll reach them!”

**Bob can help you to:**
- user-friendly
- intuitive
- 🗂️ efficient

All you need to do is,
1. download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [ ] managing tasks
- [ ] managing deadlines
- [ ] reminders

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

sarahteonin1 committed:
level 0 completed

sarahteonin1 committed:
level 0 completed

sarahteonin1 committed:
Merge remote-tracking branch 'origin/master'

sarahteonin1 committed:
level 1 completed

sarahteonin1 committed:
level 1 completed

sarahteonin1 committed:
level 1 completed

sarahteonin1 committed:
Merge tag 'Level-1'

sarahteonin1 committed:
Merge tag 'Level-1'

sarahteonin1 committed:
Merge remote-tracking branch 'origin/master'

sarahteonin1 committed:
level 2 completed

sarahteonin1 committed:
level 3 completed

sarahteonin1 committed:
level 3 completed

sarahteonin1 committed:
halfway through level 4

sarahteonin1 committed:
completed level 4

sarahteonin1 committed:
completed A-TextUiTesting

sarahteonin1 committed:
completed level 5

sarahteonin1 committed:
completed level 6

sarahteonin1 committed:
edit level 6

sarahteonin1 committed:
Add enums, switch cases to Bob.java

sarahteonin1 committed:
Add protected to all from and to

sarahteonin1 committed:
Partial implementation of FileWriting.java

sarahteonin1 committed:
Partial implementation of FileReading.java

sarahteonin1 committed:
Partial implementation of FileReading.java and FileWriting.java

sarahteonin1 committed:
Complete level-7

sarahteonin1 committed:
Add LocalDateTime to Event and Deadline Tasks

sarahteonin1 committed:
Fix minor edits in FileReading.java

sarahteonin1 committed:
Merge branch 'branch-level-7'

sarahteonin1 committed:
Add ON commands to list tasks on a specific date

sarahteonin1 committed:
Merge branch 'branch-level-8'

sarahteonin1 committed:
Complete A-MoreOOP

sarahteonin1 committed:
Delete redundant code

sarahteonin1 committed:
Merge branch 'branch-A-MoreOOP'

sarahteonin1 committed:
Complete A-Packages

sarahteonin1 committed:
Merge branch 'branch-A-Packages'

sarahteonin1 committed:
Merge branch 'add-gradle-support'

sarahteonin1 committed:
Complete JUnit Testing

sarahteonin1 committed:
Merge branch 'branch-A-JUnit'

sarahteonin1 committed:
Complete A-Jar

sarahteonin1 committed:
Merge branch 'branch-A-Jar'

sarahteonin1 committed:
Complete A-JavaDoc

sarahteonin1 committed:
Add javadoc to bob

sarahteonin1 committed:
Complete coding standards

sarahteonin1 committed:
Complete coding standards

sarahteonin1 committed:
Complete level 9

sarahteonin1 committed:
Merge branch 'branch-A-JavaDoc'

sarahteonin1 committed:
Resolve merge conflicts

sarahteonin1 committed:
Merge branch 'branch-Level-9'

sarahteonin1 committed:
Complete checkstyle for Bob package

sarahteonin1 committed:
Complete checkstyle for test cases

sarahteonin1 committed:
Merge branch-A-Checkout to master

sarahteonin1 committed:
Complete level 10

sarahteonin1 committed:
Merge branch 'branch-Level-10'

sarahteonin1 committed:
Complete level 10

gabriellegtw commented on src/main/java/bob/commands/Bye.java:
```diff
@@ -0,0 +1,37 @@
+    }
+
+    @Override
+    public boolean isExit() {
```
I like how you followed naming conventions for booleans

gabriellegtw commented on src/main/java/bob/data/BobException.java:
```diff
@@ -0,0 +1,16 @@
+ * Exception thrown by the Bob application.
+ */
+public class BobException extends Exception {
+    /**
```
I like the comprehensive javadocs used throughout the code :)

gabriellegtw commented on src/main/java/bob/storage/FileReading.java:
```diff
@@ -0,0 +1,80 @@
+        super(filePath);
+    }
+
+    protected static void createDirectory(String dirName) {
```
The way you name your methods is very clear and accurately describes its use

gabriellegtw commented on src/main/java/bob/storage/FileReading.java:
```diff
@@ -0,0 +1,80 @@
+
+    protected static TaskList loadTasks(String filePath) throws FileNotFoundException {
+        TaskList list = new TaskList();
+        File f = new File(filePath);
```
Perhaps you may want to use more descriptive variable names?

gabriellegtw reviewed (commented):
LGTM. Not much to change.

LinkesV commented on src/main/java/bob/parser/Parser.java:
```diff
@@ -0,0 +1,214 @@
+                    + " list, unmark, mark, delete, on, todo, deadline, event, and bye.");
+        }
+
+        switch (commandType) {
```
I like the way that you have incorporated enums into your code

LinkesV commented on src/main/java/bob/storage/FileReading.java:
```diff
@@ -0,0 +1,80 @@
+/**
+ * Class representing the file reading.
+ */
+public class FileReading extends Storage {
```
This class can be renamed to FileReader so that it follows the coding conventions of Class names being nouns

LinkesV commented on src/main/java/bob/storage/FileReading.java:
```diff
@@ -0,0 +1,80 @@
+        return list;
+    }
+
+    protected static boolean isDone(int i) {
```
It is good that all your boolean methods/variables have names that sound like booleans

LinkesV commented on src/test/java/bob/tasks/DeadlineTaskTest.java:
```diff
@@ -0,0 +1,41 @@
+        assertEquals("[D][ ] do assignment (by: 15 September 2024 12:00am)", new DeadlineTask("do assignment",
+                LocalDate.of(2024, 9, 15).atStartOfDay()).toString());
+    }
+}
```
It is good that you have done extensive JUnit testing for most of your methods. I have learnt a lot of things on how to write test methods from reading your code

LinkesV commented on src/main/java/bob/storage/FileReading.java:
```diff
@@ -0,0 +1,80 @@
+                LocalDateTime to = LocalDateTime.parse(parts[4]);
+                Task event = new EventTask(parts[2], from, to);
+                event.setIsDone(isDone(Integer.parseInt(parts[1])));
+                list.add(event);
```
A possible suggestion here will be for logical units to be separated by one blank line. This will enchance readability between the different segments

LinkesV reviewed (commented):
Your code follows the coding standard very closely. It looks very good

sarahteonin1 committed:
Add assertions for input validation and task formatting

This commit introduces assertions in the createDirectory, createFile, and formatTasks methods to enhance input validation and ensure that critical parameters are not null.

- In createDirectory, an assertion checks that the directory name is not null before attempting to create the directory.
- In createFile, assertions ensure the file path is not null and that the file does not already exist before creating it.
- In formatTasks, assertions validate the task type and related fields for deadline and event tasks, ensuring they meet the expected criteria.

These assertions help improve the code by catching potential issues early in the execution flow.

sarahteonin1 committed:
Refactor code to improve code quality

This commit focuses on enhancing the code quality by making the following improvements:

- Refactored methods to improve readability and reduce complexity.
- Removed redundant code and added clearer variable names.
- Improved formatting and consistency across the codebase for better maintainability.
- Updated comments and documentation to accurately reflect the changes and improve clarity.

These changes help make the code easier to understand, maintain, and extend in the future.

sarahteonin1 committed:
Add final to variables

sarahteonin1 committed:
Fix append concatenation

sarahteonin1 committed:
Change to smart switch

sarahteonin committed:
Merge pull request #1 from sarahteonin/branch-A-Assertions

Merge branch-A-Assertions with master

sarahteonin1 committed:
Resolve merge conflicts

sarahteonin committed:
Merge pull request #2 from sarahteonin/branch-A-CodeQuality

Merge branch-A-CodeQuality with master

sarahteonin1 committed:
Complete BCD-Extension

sarahteonin1 committed:
Remove redundant imports

sarahteonin1 committed:
Merge branch 'branch-BCD-Extension'

sarahteonin1 committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/bob/parser/Parser.java

sarahteonin1 committed:
Complete better gui

sarahteonin1 committed:
Complete a-personality

sarahteonin1 committed:
Fix minor bugs

sarahteonin1 committed:
Merge branch 'branch-A-Personality'

sarahteonin1 committed:
Fix ui

sarahteonin1 committed:
Complete user guide

sarahteonin1 committed:
Merge branch 'branch-A-UserGuide'

sarahteonin1 committed:
Complete user guide

sarahteonin1 committed:
Add remind to user guide

sarahteonin committed:
Update README.md

sarahteonin committed:
Update README.md
